cron/uptycs_alertpull.py

```diff
@@ -0,0 +1,208 @@
+                    self.start_time.strftime("%Y-%m-%dT%H:%M:%S.%fZ"),
+                    self.end_time.strftime("%Y-%m-%dT%H:%M:%S.%fZ")
+                ]
+            },
+            "status": {"in": ["open", "assigned"]}
+        }, sort_keys=False, separators=(',', ':'))
+
+
+class UptycsClient(object):
+    def __init__(self, path):
+        self.api_json = json.load(open(path))
+        self.url = 'https://' + self.api_json['domain'] + '.uptycs.io'
+        self.verify_ssl = True
+        self.customer_id = self.api_json['customerId']
+        self.key = self.api_json['key']
+        self.secret = self.api_json['secret']
+
+    def auth_headers(self):
+        headers = {}
+        utcnow = datetime.datetime.utcnow()
+        date = utcnow.strftime("%a, %d %b %Y %H:%M:%S GMT")
+        authVar = jwt.encode({'iss': self.key}, self.secret, algorithm='HS256')
+        authorization = "Bearer %s" % (authVar.decode('utf-8'))
+        headers['date'] = date
+        headers['Authorization'] = authorization
+        return headers
+
+    def get(self, path, params={}):
+        final_url = ("%s/public/api/customers/%s%s" %
+                     (self.url, self.customer_id, path))
+        try:
+            response = requests.get(url=final_url,
+                                    headers=self.auth_headers(),
+                                    verify=self.verify_ssl)
+            if (response and response.status_code in
+                    [requests.codes.ok, requests.codes.bad]):
+                return response.json()
+        except requests.exceptions.RequestException as e:
+            print("ERROR: {}".format(e))
+
+    def alerts(self, filter, sort='alertTime:desc&limit'):
+        path = '/alerts?filters=' + filter.filter_string() + "&sort=" + sort
+        alerts_json = self.get(path)
+        return alerts_json['items']
+
+
+def normalize(details):
+    # Normalizes fields to conform to http://mozdef.readthedocs.io/en/latest/usage.html#mandatory-fields
+    # This is mainly used for common field names to put inside the details structure
+    # There might be faster ways to do this
+    normalized = {}
+
+    # for f in details:
+    #     if f in ("ip", "ip_address", "client_ip"):
+    #         normalized["sourceipaddress"] = details[f]
+    #         continue
+    #     if f == "result":
+    #         if details[f].lower() == "success":
+    #             normalized["success"] = True
+    #         else:
+    #             normalized["success"] = False
+    #     normalized[f] = details[f]
+
+    # if "user" in normalized and type(normalized["user"]) is dict:
+    #     if "name" in normalized["user"]:
+    #         normalized["username"] = normalized["user"]["name"]
+    #     if "key" in normalized["user"]:
+    #         normalized["userkey"] = normalized["user"]["key"]
+    #     del (normalized["user"])
+
+    # return normalized
+    return details
+
+
+def process_alerts(mozmsg, uptycs_alerts):
```

Comment:
Need to have maximum alerts to MozDef threshold to shut down the run and wait until the next execution time.

Reply:
Saving this for a future iteration.